tests/test_utils.py

```diff
@@ -0,0 +1,90 @@
+# NEON AI (TM) SOFTWARE, Software Development Kit & Application Development System
+#
+# Copyright 2008-2021 Neongecko.com Inc. | All Rights Reserved
+#
+# Notice of License - Duplicating this Notice of License near the start of any file containing
+# a derivative of this software is a condition of license for this software.
+# Friendly Licensing:
+# No charge, open source royalty free use of the Neon AI software source and object is offered for
+# educational users, noncommercial enthusiasts, Public Benefit Corporations (and LLCs) and
+# Social Purpose Corporations (and LLCs). Developers can contact developers@neon.ai
+# For commercial licensing, distribution of derivative works or redistribution please contact licenses@neon.ai
+# Distributed on an "AS IS” basis without warranties or conditions of any kind, either express or implied.
+# Trademarks of Neongecko: Neon AI(TM), Neon Assist (TM), Neon Communicator(TM), Klat(TM)
+# Authors: Guy Daniels, Daniel McKnight, Elon Gasper, Richard Leeds, Kirill Hrymailo
+#
+# Specialized conversational reconveyance options from Conversation Processing Intelligence Corp.
+# US Patents 2008-2021: US7424516, US20140161250, US20140177813, US8638908, US8068604, US8553852, US10530923, US10530924
+# China Patent: CN102017585  -  Europe Patent: EU2156652  -  Patents Pending
+
+import os
+import sys
+import time
+import unittest
+import pytest
+
+sys.path.insert(0, os.path.join(os.path.dirname(__file__), ".."))
+
+from neon_mq_connector.utils import RepeatingTimer
+from neon_mq_connector.utils.connection_utils import get_timeout, retry
+
+
+def callback_on_failure():
+    """Simple callback on failure"""
+    return False
+
+
+class TestMQConnectorUtils(unittest.TestCase):
+
+    counter = 0
+
+    def repeating_method(self):
+        """Simple method incrementing counter by one"""
+        self.counter += 1
+
+    @retry(num_retries=3, backoff_factor=0.1, callback_on_exceeded=callback_on_failure, use_self=True)
+    def method_passing_on_nth_attempt(self, num_attempts: int = 3) -> bool:
+        """
+            Simple method that is passing check only after n-th attempt
+            :param num_attempts: number of attempts before passing
+        """
+        if self.counter < num_attempts-1:
+            self.repeating_method()
+            raise AssertionError('Awaiting counter equal to 3')
+        return True
+
+    def test_01_get_timeout(self):
```

Comment:
Do these 3 tests need to run synchronously, or can the numeric indices be removed from method names?